darts/dataprocessing/transformers/base_data_transformer.py

```diff
@@ -21,8 +24,38 @@ def __init__(self,
         ----------
         names
             The data transformer's name
+        n_jobs
+            The number of jobs to run in parallel. Defaults to `1`. `-1` means using all processors
+        verbose
+            Optionally, whether to print operations progress
         """
         self._name = name
+        self._verbose = verbose
+        self._n_jobs = n_jobs
+
+    def set_verbose(self, value: bool):
+        """
+        Setter for the verbosity status. True for enabling the datailed report about scaler's operation progress, False
+        for no additional information
+
+        Parameters
+        ----------
+        value
+            New verbosity status
+
+        """
+        raise_if_not(isinstance(value, bool), "Verbosity status must be a boolean.")
+
+        self._verbose = value
+
+    def set_n_jobs(self, value: int):
+        """
+        Sets the number of cores that will be used by the transformer while processing multiple time series. Set to `-1` 
+        for using all the available cores.
+        """
+
+        raise_if_not(isinstance(value, int), "n_jobs must be an integer")
+        self._n_jobs = value
 
```

Comment:
In the transformers you often have a pattern like that:

```
def transform():
        ...

        if isinstance(data, TimeSeries):
            return some_fn(data)
        else:
            def map_ts(ts):
                return some_fn(ts)

            iterator = _build_tqdm_iterator(data)

            transformed_data = Parallel(n_jobs=self._n_jobs)(delayed(map_ts)(ts) for ts in iterator)
```

This leads to quite a bit of code duplication, because all transformers now have to implement the (same) parallelism code. How about having the `Parallel` business done only in the super class(es)? Instead of `transform()` and `inverse_transform()` being abstract, you could have them call an abstract method of the form
```
@abstract
def ts_transformation(ts: TimeSeries) -> TimeSeries
    pass
    
@abstract
def ts_inverse_transformation(ts: TimeSeries) -> TimeSeries
    pass
```
They would do the same thing as the "old" transform & inverse_transform methods, but be called by the new transform & inverse_transform methods, which would no longer be abstract and provide parallelism.